### PR TITLE
Do not raise an exception on a faulty Auth Header

### PR DESCRIPTION
**Pull request by kfrohwein**
## PR Description

Currently if the token read from the header contains no spaces there will be an exception.
But this should only mean that you can't authorize with this header.
So I now check if the preg_split is successful and if no we just retun an empty
Permission Collection Transfer in the end.

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: src/Spryker/Client/OauthPermission/OauthPermission/OauthPermissionReader.php [hunk 1]
```diff
@@ -85,10 +85,16 @@ protected function extractPermissionsFromOauthToken(OauthAccessTokenDataTransfer
     /**
      * @param string $authorizationToken
      *
-     * @return array
+     * @return string
```
Comment (dereuromark) [inline src/Spryker/Client/OauthPermission/OauthPermission/OauthPermissionReader.php]:
```suggestion
     * @return string|null
```
based on your changes.

File: src/Spryker/Client/OauthPermission/OauthPermission/OauthPermissionReader.php [hunk 2]
```diff
@@ -47,12 +47,12 @@ public function getPermissionsFromOauthToken(): PermissionCollectionTransfer
         $request = Request::createFromGlobals();
         $authorizationToken = $request->headers->get(OauthPermissionConfig::HEADER_AUTHORIZATION);
 
-        if (!$authorizationToken) {
+        $accessToken = $this->extractToken($authorizationToken);
```
Comment (dereuromark) [inline src/Spryker/Client/OauthPermission/OauthPermission/OauthPermissionReader.php:50]: Could this method get null passed now and break? Looks like it.

Comment (kfrohwein) [reply src/Spryker/Client/OauthPermission/OauthPermission/OauthPermissionReader.php:50]: You are fast ;) Yes that's right.

Comment (kfrohwein) [reply src/Spryker/Client/OauthPermission/OauthPermission/OauthPermissionReader.php:50]: Maybe like this? And I should add a test. But I currently have no valilla spryker setup to test this stand alone.

File: src/Spryker/Client/OauthPermission/OauthPermission/OauthPermissionReader.php [hunk 3]
```diff
@@ -85,10 +85,20 @@ protected function extractPermissionsFromOauthToken(OauthAccessTokenDataTransfer
     /**
      * @param string $authorizationToken
```
Comment (dereuromark) [inline src/Spryker/Client/OauthPermission/OauthPermission/OauthPermissionReader.php]: This is also not quite correct now :)
```suggestion
     * @param string|null $authorizationToken
```

**Comment by SriRamyaVadlamudi** [comment]
Hi @kfrohwein thank you for your contribution into Spryker! 
Due to our development and release process (we develop in an internal monorepo), I will create an internal ticket for this PR so our development team will import it (you will remain as an author though) and after QA and architecture review this change will be released back from our internal repo to this one. 
I will notify when it happened. Unfortunately, due to limited capacity this can take some undefined amount of time. 

Have a nice day! 

Best, 
Spryker Support Team

File: src/Spryker/Client/OauthPermission/OauthPermission/OauthPermissionReader.php [hunk 4]
```diff
@@ -83,12 +83,22 @@ protected function extractPermissionsFromOauthToken(OauthAccessTokenDataTransfer
...
+     * @param string|null $authorizationToken
      *
-     * @return array
+     * @return string|null
      */
-    protected function extractToken(string $authorizationToken): array
+    protected function extractToken(?string $authorizationToken): ?string
     {
-        return preg_split('/\s+/', $authorizationToken);
+        if (empty($authorizationToken)) {
```
Comment (dereuromark) [inline src/Spryker/Client/OauthPermission/OauthPermission/OauthPermissionReader.php:92]:
```suggestion
        if (!$authorizationToken) {
```
is per new coding standard

@SriRamyaVadlamudi Can we please schedule this as part of our bugfix process?

**Comment by dimitriyTsemma** [comment]
Hi @kfrohwein thank you for your contribution into Spryker!
this PR has conflicts, please resolve them.

**Comment by dimitriyTsemma** [comment]
Dear contributor, the pull request you suggested has been successfully processed and released within the `oauth-permission` version: 1.4.0. Thank you very much for your contribution to the development of Spryker and the formation of the Spryker community!

**Pull request closed without merging**